packages/common/src/stream.ts

```diff
@@ -249,7 +251,7 @@ export abstract class Stream extends Observable<StreamState> implements StreamSt
   }
 
   get state(): StreamState {
-    return cloneDeep(this.state$.value)
```

Comment:
this is a potentially a quite consequential change.  I'm open to making it, but can we separate it out of this PR into its own more targeted PR where we can discuss the implications in more depth without holding up the rest of the changes here?